include/deal.II/cgal/triangulation.h

```diff
@@ -0,0 +1,403 @@
+   * polyhedral mesh. Generally speaking, a triangulation is made of simplices,
+   * whereas a polygonal or polyhedral mesh is made of general polygons or
+   * general polyhedrons. While CGAL implements the two concepts in a similar
+   * fashion using half-edge data structures, some optimizations are performed
+   * for triangulations, where only triangles or tetrahedra are used.
+   *
+   * @param[in] points The input points to build the triangulation from.
+   * @param[out] triangulation The output triangulation.
+   */
+  template <int spacedim, typename CGALTriangulation>
+  void
+  add_points_to_cgal_triangulation(const std::vector<Point<spacedim>> &points,
+                                   CGALTriangulation &triangulation);
+
+  /**
+   * Convert any compatible CGAL triangulation type to a deal.II triangulation.
+   *
+   * The conversion is done by looping over the finite vertices and finite cells
+   * of the CGAL triangulation, and building with them a deal.II Triangulation
+   * object.
+   *
+   * CGAL considers triangulations to be a partition of the entire
+   * spacedim-dimensional space, where every face (including boundary faces) is
+   * shared between neighboring cells. This is obtained by adding to the list of
+   * vertices a special vertex (the vertex at infinity, representing a point on
+   * the spacedim-dimensional sphere with infinite radius), and adding a
+   * neighboring infinite cell to each boundary face.
+   *
+   * Valid conversions require that the capacity of the deal.II triangulation
+   * matches that of the input CGAL triangulation, i.e., if the input CGAL
+   * triangulation is a CGAL::Triangulation_2, then spacedim must be greater
+   * or equal than two. If the input CGAL triangulation is a
+   * CGAL::Triangulation_3, then spacedim must equal than three.
+   *
+   * Both CGAL::Triangulation_2 and CGAL::Triangulation_3 can store degenerate
+   * two dimensional triangulations (i.e., a one-dimensional triangulation), or
+   * a degenerate three dimensional triangulation (i.e., a two-dimensional
+   * surface triangulation). The dimension of the triangulation, as returned by
+   * `cgal_triangulation.dimension()`, must match the dimension `dim` of the
+   * input Triangulation `dealii_triangulation`. If this is not the case, an
+   * exception is thrown.
+   *
+   * @param cgal_triangulation
+   * @param dealii_triangulation
+   */
+  template <typename CGALTriangulation, int dim, int spacedim>
+  void
+  cgal_triangulation_to_dealii_triangulation(
+    const CGALTriangulation &     cgal_triangulation,
+    Triangulation<dim, spacedim> &dealii_triangulation);
+
+  /**
+   * @brief Construct a deal.II triangulation starting from a CGAL_MeshType object,
+   * which can either be a CGAL::Surface_mesh or a CGAL::Polyhedron_3. These
+   * types can both represent a polyhedral surface. Therefore, the natural
+   * deal.II object to fill with this function is a Triangulation<2,3>.
```

Comment:
Since we are not allowing any other types of deal.II triangulation, we might as well remove the dim and spacedim parameters, unless you are planning to use the same function to create a volumetric triangulation from a surface mesh.